### ISSUE TITLE
Support prefix / overlapping auto-closing bracket pairs

### ISSUE THREAD

**Issue by butuzov** [microsoft/vscode#102552]
## TL/DR 

VSCode autocompletes based on language configuration, but checking chars already types and results are extra closing brackets.

## Report Issue to prefill these.  
- VSCode Version: 1.47
- OS Version: macOS 10.14

## Steps to Reproduce:

1. Install Jinja extension (https://github.com/wholroyd/vscode-jinja or https://marketplace.visualstudio.com/items?itemName=wholroyd.jinja )
2. Ensure that `Editor: Autoclosing Brackets` is set to `languageDefined` (there were no tests with other options)
3. Create Jinja file (test.html.j2)
4. Type `{` 
    * VScode autocomplete to `{}`
5. Type `%` 
    * VSCode find opening `{%`
    * VSCode finds matched entry in `brackets` of [jinja.configuration.json](https://github.com/wholroyd/vscode-jinja/blob/master/jinja.configuration.json) - `{%`
    * VSCode add autoclose entry `%}`
    * Result string `{%%}}`
    
## Does this issue occur when all extensions are disabled?: Yes/No
No, as it depends on external configuration.

**Comment by alexdima** [comment]
The relevant part in the language configuration is:
```
{
	"comments": {
		"lineComment": "#"
	},
	"brackets": [
		["{", "}"],
		["[", "]"],
		["(", ")"],
		["{%", "%}"]
	]
}
```

**Comment by butuzov** [comment]
HI Alexanderu

That (language configuration) I was talking about (`contributes/extensions/configuration`. I also tried to come up with any other engine that has similar problems, but looks Jinja only one with such overlapping prefixes issue).

**Comment by butuzov** [comment]
I think its can be closed in favor of the microsoft/vscode#72177 (found via tag you assigned). What do you think?

**Comment by alexdima** [comment]
Let's track in  #72177
